client/src/components/edit-instance-view/EditInstanceForm.js

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import Grid2 from '@mui/material/Unstable_Grid2';
+import textFormatServices from '../../services/textFormat';
+import DynamicTextFieldArray from './DynamicTextFieldArray';
+import StringTextField from './StringTextFiled';
+import NumberTextField from './NumberTextField';
+
+// TODO: form handling 
+// e.g. check that numbers, dates and emails are submited and nothing of the wrong format
+
+const typeData = {
+  fieldId: 'instanceType',
+  fieldLabel: 'Type'
+};
+
+const startDateData = {
+  fieldId: 'startDate',
+  fieldLabel: 'Starting Date'
+};
+
+const endDateData = {
+  fieldId: 'endDate',
+  fieldLabel: 'Ending Date'
+};
+
+const minCreditsData = {
+  fieldId: 'minCredits',
+  fieldLabel: 'Min Credits'
+};
+
+const maxCreditsData = {
+  fieldId: 'maxCredits',
+  fieldLabel: 'Max Credits'
+};
+
+const gradingScaleData = {
+  fieldId: 'gradingScale',
+  fieldLabel: 'Grading Scale'
+};
+
+const teacherData = {
+  fieldId: 'teacher',
+  fieldLabel: 'Instance Teacher'
```

Comment:
"Teacher of Instance" may be better.

Reply:
Changed the label to "Teacher of This Instance" since it seemed clearer.